core/src/main/java/fr/sncf/osrd/standalone_sim/StandaloneSim.java

```diff
@@ -156,4 +164,49 @@ public static Envelope applyAllowances(
         }
         return result;
     }
+
+    /**  Generate a mareco Allowance given a list of scheduled points.
+     *   Return an empty value:
+     *   - if no scheduled point is given
+     *   - if we cannot respect any scheduled points. */
+    public static Optional<MarecoAllowance> generateAllowanceFromScheduledPoints(
+            Envelope maxEffortEnvelope,
+            ArrayList<ScheduledPoint> scheduledPoints
+    ) {
+        scheduledPoints.sort(Comparator.comparingDouble(sp -> sp.position));
+        var ranges = new ArrayList<AllowanceRange>();
+        double rangeBeginPos = 0.;
+        double lostTime = 0.;
+        for (var schedulePoint : scheduledPoints) {
+            var rangeEndPos = schedulePoint.position;
+            double excessTime = schedulePoint.time - maxEffortEnvelope.interpolateTotalTime(rangeEndPos) - lostTime;
```

Comment:
I believe maxEffortEnvelope does not take stop times into account when interpolating time. This is unfortunate, and while we would like to correct for this drawback in the future, it means that you need to use EnvelopeStopWrapper in order to properly interpolate time, including stop duration.

Reply:
Good catch. It's now fixed.